### ISSUE TITLE
Panic with a single node deployment without auth or tls

### ISSUE THREAD

**Issue by Vilsol** [arangodb/kube-arangodb#639]
The arango operator keeps crashlooping with this error:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x11fca57]

goroutine 104 [running]:
github.com/arangodb/go-driver/http.(*httpConnection).SetAuthentication(0xc00030ba40, 0x0, 0x0, 0xc00032ef20, 0xc000d0ccc8, 0x11fa82d, 0x0)
	/home/ubuntu/.GIT/arango/kube-arangodb/.gobuild/pkg/mod/github.com/arangodb/go-driver@v0.0.0-20200617115956-9dac4c7fed22/http/connection.go:389 +0x37
github.com/arangodb/go-driver/cluster.(*clusterConnection).SetAuthentication(0xc0001e6e00, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/ubuntu/.GIT/arango/kube-arangodb/.gobuild/pkg/mod/github.com/arangodb/go-driver@v0.0.0-20200617115956-9dac4c7fed22/cluster/cluster.go:297 +0x121
github.com/arangodb/kube-arangodb/pkg/util/arangod/conn.factory.Connection(0xc000aac1c0, 0xc000aac1d0, 0xc00032eed0, 0x1, 0x1, 0xc000d0ce20, 0x4, 0x8, 0x10)
	/home/ubuntu/.GIT/arango/kube-arangodb/pkg/util/arangod/conn/factory.go:134 +0x15e
github.com/arangodb/kube-arangodb/pkg/util/arangod/conn.factory.Client(0xc000aac1c0, 0xc000aac1d0, 0xc00032eed0, 0x1, 0x1, 0x10, 0x1633360, 0x1, 0xc00032eed0)
	/home/ubuntu/.GIT/arango/kube-arangodb/pkg/util/arangod/conn/factory.go:84 +0x66
github.com/arangodb/kube-arangodb/pkg/deployment.(*clientCache).getDatabaseClient(0xc00037a240, 0x0, 0x0, 0x0, 0x0)
	/home/ubuntu/.GIT/arango/kube-arangodb/pkg/deployment/client_cache.go:120 +0xf0
github.com/arangodb/kube-arangodb/pkg/deployment.(*clientCache).getDatabase(0xc00037a240, 0x1dd9fc0, 0xc0001aa000, 0x0, 0x0, 0x0, 0x0)
	/home/ubuntu/.GIT/arango/kube-arangodb/pkg/deployment/client_cache.go:129 +0x40
github.com/arangodb/kube-arangodb/pkg/deployment.(*clientCache).GetDatabase(0xc00037a240, 0x1dd9fc0, 0xc0001aa000, 0x0, 0x0, 0x0, 0x0)
	/home/ubuntu/.GIT/arango/kube-arangodb/pkg/deployment/client_cache.go:150 +0x8c
github.com/arangodb/kube-arangodb/pkg/deployment.(*Deployment).runBootstrap(0xc0003c9800, 0x18c4431, 0x17)
	/home/ubuntu/.GIT/arango/kube-arangodb/pkg/deployment/bootstrap.go:129 +0x78
github.com/arangodb/kube-arangodb/pkg/deployment.(*Deployment).EnsureBootstrap(0xc0003c9800, 0x0, 0x0)
	/home/ubuntu/.GIT/arango/kube-arangodb/pkg/deployment/bootstrap.go:53 +0x37c
github.com/arangodb/kube-arangodb/pkg/deployment.(*Deployment).inspectDeploymentWithError(0xc0003c9800, 0x1dd9fc0, 0xc0001aa000, 0x2540be400, 0x1dffbc0, 0xc000313c70, 0x2540be400, 0x0, 0x0)
	/home/ubuntu/.GIT/arango/kube-arangodb/pkg/deployment/deployment_inspector.go:261 +0x9bc
github.com/arangodb/kube-arangodb/pkg/deployment.(*Deployment).inspectDeployment(0xc0003c9800, 0x2540be400, 0x0)
	/home/ubuntu/.GIT/arango/kube-arangodb/pkg/deployment/deployment_inspector.go:102 +0x506
github.com/arangodb/kube-arangodb/pkg/deployment.(*Deployment).run(0xc0003c9800)
	/home/ubuntu/.GIT/arango/kube-arangodb/pkg/deployment/deployment.go:272 +0x5e9
created by github.com/arangodb/kube-arangodb/pkg/deployment.New
	/home/ubuntu/.GIT/arango/kube-arangodb/pkg/deployment/deployment.go:154 +0x577
```

This begins when after I have deployed this:
```yaml
apiVersion: database.arangodb.com/v1alpha
kind: ArangoDeployment
metadata:
  name: arangodb
spec:
  mode: Single
  auth:
    jwtSecretName: None
  externalAccess:
    type: None
  tls:
    caSecretName: None
```

I am running latest minikube (1.13.1) on the docker driver using kubernetes version 1.19.2.

I installed arango operator via helm chart version 1.0.7

**Comment by ajanikow** [comment]
Hello!

I will investigate this problem and share with you details. 

Best Regards,
Adam.

**Comment by kormotodor** [comment]
Same

**Comment by ajanikow** [comment]
Fixed in 1.0.9 and 1.1.0
